### PR TITLE
Workbench/MantidPlot: Project Recovery - Remove file dumping from the logging

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
Everytime project recovery ran in Workbench/MantidPlot and Instrument View was open it would dump the masked workspace.

**To test:**
- Open MantidPlot
- Load a workspace with an instrument
- Open Instrument View
- Run mantidplot.app.saveRecoveryCheckpoint()
- It should not dump something similar to the following, in the console:
```
<detector-masking>
    <group>
        <detids/>
    </group>
</detector-masking>
```

Testing on MantidPlot should be sufficient as both use the same code (Feel free to repeat on Workbench, you will have to wait for recovery).

*There is no associated issue.*

It's already got a suitable thing in the MantidPlot release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Kernel/inc/MantidKernel/DataService.h [hunk 1]
```diff
@@ -287,7 +287,7 @@ template <typename T> class DLLExport DataService {
     lock.unlock();
     notificationCenter.postNotification(new PreDeleteNotification(name, data));
     data.reset(); // DataService now has no references to the object
-    g_log.information("Data Object '" + name + "' deleted from data service.");
+    g_log.debug("Data Object '" + name + "' deleted from data service.");
```
Comment (martyngigg) [inline Framework/Kernel/inc/MantidKernel/DataService.h]: While we are changing this to be consistent with the other methods. Can you also change the `.information` in `rename`?

**Review by martyngigg — CHANGES_REQUESTED** [review]
One minor request before I start testing.

File: Framework/Kernel/inc/MantidKernel/DataService.h (hunk 1 above)
Comment (Pasarus) [reply Framework/Kernel/inc/MantidKernel/DataService.h]: Done :+1:

**Review by martyngigg — APPROVED** [review]
The changes are sensible and remove a spurious message from the project saving code.

**Pull request merged**